ts/features/idpay/unsubscription/xstate/services.ts

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable no-underscore-dangle */
+import * as E from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/function";
+import { PreferredLanguage } from "../../../../../definitions/backend/PreferredLanguage";
+import { InitiativeDTO } from "../../../../../definitions/idpay/InitiativeDTO";
+import { IDPayClient } from "../../common/api/client";
+import { Context } from "./context";
+
+export type Services = {
+  getInitiativeInfo: {
+    data: InitiativeDTO;
+  };
+  unsubscribeFromInitiative: {
+    data: undefined;
+  };
+};
+
+const createServicesImplementation = (
+  client: IDPayClient,
+  token: string,
+  language: PreferredLanguage
+) => {
+  const getInitiativeInfo = async (context: Context) => {
+    try {
+      const dataResponse = await client.getWalletDetail({
+        bearerAuth: token,
+        "Accept-Language": language,
+        initiativeId: context.initiativeId
+      });
+
+      const data: Promise<InitiativeDTO> = pipe(
+        dataResponse,
+        E.fold(
+          _ => Promise.reject(),
+          response => {
+            if (response.status !== 200) {
+              return Promise.reject();
+            }
+            return Promise.resolve(response.value);
+          }
+        )
+      );
+
+      return data;
+    } catch {
+      return Promise.reject();
+    }
+  };
+
+  const unsubscribeFromInitiative = async (context: Context) => {
+    try {
+      const dataResponse = await client.unsubscribe({
+        bearerAuth: token,
+        "Accept-Language": language,
+        initiativeId: context.initiativeId
+      });
+
+      const data: Promise<undefined> = pipe(
+        dataResponse,
+        E.fold(
+          _ => Promise.reject(undefined),
+          response => {
+            if (response.status !== 200) {
+              return Promise.reject(undefined);
+            }
+            return Promise.resolve(undefined);
+          }
```

Comment:
```suggestion
          response => response.status !== 200 ? Promise.reject(undefined) : Promise.resolve(undefined)
```

Reply:
Refactored in https://github.com/pagopa/io-app/pull/4462/commits/b67896918ada02b13d8a437e87bc1aac3ed991ec